File: Python/7py 3/solve_srd.py
"""
|--------------------------------------------------------------------------
| Simple remove duplicates
|--------------------------------------------------------------------------
|
"""

def solve(arr): 
   return [v for i, v in enumerate(arr) if v not in arr[i+1:]]


def solve(arr): 
   return list(dict.fromkeys(arr[::-1]))[::-1]


def solve(arr): 
   return sorted(set(arr), key=arr[::-1].index)[::-1]


print(solve([3,4,4,3,6,3]),[4,6,3])
print(solve([1,2,1,2,1,2,3]),[1,2,3])
print(solve([1,2,3,4]),[1,2,3,4])
print(solve([1,1,4,5,1,2,1]),[4,5,2,1])

File: PHP/7php 3/solve_srd.php
<?php
/*
|--------------------------------------------------------------------------
| Simple remove duplicates
|--------------------------------------------------------------------------
|
*/

function solve($arr) {
   return array_reverse(array_unique(array_reverse($arr)));
}


function solve($arr) {
   $res = [];
   foreach ($arr as $k => $v) {    
      if (!in_array($v, array_slice($arr,$k+1))) $res[] = $v;
   }
   return $res;
}


function solve($arr) {
   $arr = array_flip(array_flip($arr));
   ksort($arr);
   return array_values($arr);
}


print_r(solve([3, 4, 4, 3, 6, 3]));
print_r(solve([1, 2, 1, 2, 1, 2, 3]));